engine/class_modules/sc_shaman.cpp

```diff
@@ -5673,6 +5668,10 @@ struct flame_shock_t : public shaman_spell_t
     {
       proc_chance = p()->azerite.igneous_potential.spell_ref().effectN( 3 ).percent();
     }
+    if ( p()->spec.restoration_shaman->ok() )
+    {
+      proc_chance += p()->spec.restoration_shaman->effectN( 8 ).percent();
+    }
```

Comment:
While this results in the correct value, I'd add the actual `Igneous Potential` effect 4 for restoration, too. 

```cpp
    proc_chance = p()->spec.lava_surge->effectN( 1 ).percent();
    
    if ( p()->spec.restoration_shaman->ok() )
    {
      proc_chance += p()->spec.restoration_shaman->effectN( 8 ).percent();
    }

    if ( p()->azerite.igneous_potential.ok() )
    {
      if ( p()->spec.restoration_shaman->ok() ) {
        proc_chance = p()->azerite.igneous_potential.spell_ref().effectN( 4 ).percent();
      } else if ( p()->spec.elemental_shaman->ok() {
        proc_chance = p()->azerite.igneous_potential.spell_ref().effectN( 3 ).percent();
      }
    }
```